From 7c112f5c613cfc8a93ca3d85ccdab7a38b82a3c3 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@qt.io>
Date: Thu, 3 Feb 2022 15:03:22 +0100
Subject: [PATCH] Find GIO with QtBase 6.2

GLIB2 doesn't find GIO with Qt 6.2, so try adding a fallback.

Pick-to: 6.3
Task-number: QTBUG-100435
Change-Id: Ic98a7aea07b726dc4f6e76a65669ca92b1916b5c
Reviewed-by: Kirill Burtsev <kirill.burtsev@qt.io>
---
 src/core/api/configure.cmake | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/core/api/configure.cmake b/src/core/api/configure.cmake
index 231eaa8d2..a9dd9eb09 100644
--- a/src/core/api/configure.cmake
+++ b/src/core/api/configure.cmake
@@ -8,6 +8,9 @@ if(NOT QT_CONFIGURE_RUNNING)
         pkg_check_modules(ALSA alsa IMPORTED_TARGET)
         pkg_check_modules(PULSEAUDIO libpulse>=0.9.10 libpulse-mainloop-glib)
         pkg_check_modules(XDAMAGE xdamage)
+        if(NOT GIO_FOUND)
+            pkg_check_modules(GIO gio-2.0)
+        endif()
     endif()
 
     find_package(Qt6 ${PROJECT_VERSION} CONFIG QUIET
@@ -111,7 +114,7 @@ qt_feature("webengine-webrtc-pipewire" PRIVATE
     LABEL "PipeWire over GIO"
     PURPOSE "Provides PipeWire support in WebRTC using GIO."
     AUTODETECT false
-    CONDITION QT_FEATURE_webengine_webrtc AND TARGET GLIB2::GIO
+    CONDITION QT_FEATURE_webengine_webrtc AND GIO_FOUND
 )
 qt_feature("webengine-full-debug-info" PRIVATE
     SECTION "WebEngine"
-- 
2.35.1

